splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/CrossJoinStrategy.java

```diff
@@ -278,10 +278,10 @@ public static double crossJoinStrategyLocalCost(CostEstimate innerCost, CostEsti
         assert outerCost.getRemoteCostPerPartition() != 0d || outerCost.remoteCost() == 0d;
         double innerLocalCost = innerCost.getLocalCostPerPartition()*innerCost.partitionCount();
         double innerRemoteCost = innerCost.getRemoteCostPerPartition()*innerCost.partitionCount();
+        // the cost is similar to broadcast join except that the joingRowCost is larger as it is lack of equality conditions
         return outerCost.getLocalCostPerPartition() +
-                outerCost.getRemoteCostPerPartition()  +
-                (outerCost.rowCount()/outerCost.partitionCount()) * (innerLocalCost + innerRemoteCost) +
-                joiningRowCost/outerCost.partitionCount();
+                (innerLocalCost + innerRemoteCost) +innerCost.getOpenCost()+innerCost.getCloseCost()+.01 +
```

Comment:
Is the + 0.01 there to differentiate from broadcast join?
If so, we'd have to be careful to keep them in sync

Reply:
No, the +0.01 is copied over from broadcast join costing.